src/minesweeper.rs

```diff
@@ -288,10 +288,8 @@ impl Minesweeper {
 
     fn generate_mines(&mut self, num_mines: i32, exclude_x: i32, exclude_y: i32) {
         self.mines.clear();
-        for _x in 0..self.game_board_width {
-            for _y in 0..self.game_board_height {
-                self.mines.push(false);
-            }
+        for _ in 0..self.game_board_width * self.game_board_height {
+            self.mines.push(false);
```

Comment:
You should be able to replace both `clear` and the entire loop with `self.mines.resize(width * height, false)`.

Reply:
it seems like resize only affects the items that will be appended to the vector if the desired size is larger than the current one, maybe setting the size in `new` and iterating through the vectors with mutable references with which we set default values will be more efficient